### ISSUE TITLE
Unable to install or build PAT on MacOS High Sierra (10.13.2)

### ISSUE THREAD

**Issue by mouse07410** [la5nta/pat#116]
I installed `go` via Macports (it installs stuff to `/opt/local`)  and set `export GOROOT=/opt/local/lib/go`.

Then I created directory `$HOME/src/gosrc` and did `export GOPATH=$HOME/src/gosrc`. Then I did `cd $GOPATH; go get -u github.com/la5nta/pat`. Result:
```
$ go get -u github.com/la5nta/pat
package github.com/la5nta/pat/vendor/github.com/bndr/gotabulate: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/bndr/gotabulate/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/elazarl/go-bindata-assetfs: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/elazarl/go-bindata-assetfs/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/fsnotify/fsnotify: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/fsnotify/fsnotify/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/gorhill/cronexpr: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/gorhill/cronexpr/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/gorilla/mux: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/gorilla/mux/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/gorilla/websocket: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/gorilla/websocket/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/howeyc/gopass: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/howeyc/gopass/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/transport/ardop: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/fbb/vendor/github.com/paulrosania/go-charset/charset: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/fbb/vendor/github.com/paulrosania/go-charset/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/transport/ax25/vendor/github.com/tarm/goserial: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/transport/ax25/vendor/github.com/tarm/goserial/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/microcosm-cc/bluemonday: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/microcosm-cc/bluemonday/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/nfnt/resize: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/nfnt/resize/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/peterh/liner: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/peterh/liner/.git exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/spf13/pflag: /Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/spf13/pflag/.git exists but is not a directory
$ 
```

Attempt to manually clone and build also failed:
```
$ go env gopath
$ echo $GOROOT
/opt/local/lib/go
$ echo $GOPATH
/Users/uri/src/gosrc
$ pwd
/Users/uri/src/gosrc
$ ls
src/
$ git clone https://github.com/la5nta/pat
Cloning into 'pat'...
remote: Counting objects: 2817, done.
remote: Total 2817 (delta 0), reused 0 (delta 0), pack-reused 2817
Receiving objects: 100% (2817/2817), 4.14 MiB | 1.41 MiB/s, done.
Resolving deltas: 100% (1633/1633), done.
$ cd pat
$ ./make.bash 
Updating git submodules...
Submodule 'vendor/github.com/bndr/gotabulate' (https://github.com/bndr/gotabulate.git) registered for path 'vendor/github.com/bndr/gotabulate'
Submodule 'vendor/github.com/elazarl/go-bindata-assetfs' (https://github.com/elazarl/go-bindata-assetfs.git) registered for path 'vendor/github.com/elazarl/go-bindata-assetfs'
Submodule 'vendor/github.com/fsnotify/fsnotify' (https://github.com/fsnotify/fsnotify.git) registered for path 'vendor/github.com/fsnotify/fsnotify'
Submodule 'vendor/github.com/gorhill/cronexpr' (https://github.com/gorhill/cronexpr.git) registered for path 'vendor/github.com/gorhill/cronexpr'
Submodule 'vendor/github.com/gorilla/context' (https://github.com/gorilla/context.git) registered for path 'vendor/github.com/gorilla/context'
Submodule 'vendor/github.com/gorilla/mux' (https://github.com/gorilla/mux.git) registered for path 'vendor/github.com/gorilla/mux'
Submodule 'vendor/github.com/gorilla/websocket' (https://github.com/gorilla/websocket.git) registered for path 'vendor/github.com/gorilla/websocket'
Submodule 'vendor/github.com/howeyc/gopass' (https://github.com/howeyc/gopass.git) registered for path 'vendor/github.com/howeyc/gopass'
Submodule 'vendor/github.com/jteeuwen/go-bindata' (https://github.com/jteeuwen/go-bindata.git) registered for path 'vendor/github.com/jteeuwen/go-bindata'
Submodule 'vendor/github.com/la5nta/wl2k-go' (https://github.com/la5nta/wl2k-go) registered for path 'vendor/github.com/la5nta/wl2k-go'
Submodule 'vendor/github.com/mattn/go-runewidth' (https://github.com/mattn/go-runewidth.git) registered for path 'vendor/github.com/mattn/go-runewidth'
Submodule 'vendor/github.com/microcosm-cc/bluemonday' (https://github.com/microcosm-cc/bluemonday.git) registered for path 'vendor/github.com/microcosm-cc/bluemonday'
Submodule 'vendor/github.com/nfnt/resize' (https://github.com/nfnt/resize.git) registered for path 'vendor/github.com/nfnt/resize'
Submodule 'vendor/github.com/peterh/liner' (https://github.com/peterh/liner.git) registered for path 'vendor/github.com/peterh/liner'
Submodule 'vendor/github.com/spf13/pflag' (https://github.com/spf13/pflag.git) registered for path 'vendor/github.com/spf13/pflag'
Submodule 'vendor/golang.org/x/crypto' (https://go.googlesource.com/crypto) registered for path 'vendor/golang.org/x/crypto'
Submodule 'vendor/golang.org/x/net' (https://go.googlesource.com/net) registered for path 'vendor/golang.org/x/net'
Submodule 'vendor/golang.org/x/sys' (https://go.googlesource.com/sys) registered for path 'vendor/golang.org/x/sys'
Submodule 'vendor/golang.org/x/text' (https://go.googlesource.com/text) registered for path 'vendor/golang.org/x/text'
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/bndr/gotabulate'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/elazarl/go-bindata-assetfs'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/fsnotify/fsnotify'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/gorhill/cronexpr'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/gorilla/context'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/gorilla/mux'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/gorilla/websocket'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/howeyc/gopass'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/jteeuwen/go-bindata'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/la5nta/wl2k-go'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/mattn/go-runewidth'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/microcosm-cc/bluemonday'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/nfnt/resize'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/peterh/liner'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/spf13/pflag'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/golang.org/x/crypto'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/golang.org/x/net'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/golang.org/x/sys'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/golang.org/x/text'...
Submodule path 'vendor/github.com/bndr/gotabulate': checked out 'bc555436bfd5d4e93a2dd35b462ce37e067afdda'
Submodule path 'vendor/github.com/elazarl/go-bindata-assetfs': checked out '30f82fa23fd844bd5bb1e5f216db87fd77b5eb43'
Submodule path 'vendor/github.com/fsnotify/fsnotify': checked out '4da3e2cfbabc9f751898f250b49f2439785783a1'
Submodule path 'vendor/github.com/gorhill/cronexpr': checked out 'd520615e531a6bf3fb69406b9eba718261285ec8'
Submodule path 'vendor/github.com/gorilla/context': checked out '08b5f424b9271eedf6f9f0ce86cb9396ed337a42'
Submodule path 'vendor/github.com/gorilla/mux': checked out 'ac112f7d75a0714af1bd86ab17749b31f7809640'
Submodule path 'vendor/github.com/gorilla/websocket': checked out 'a69d9f6de432e2c6b296a947d8a5ee88f68522cf'
Submodule path 'vendor/github.com/howeyc/gopass': checked out 'bf9dde6d0d2c004a008c27aaee91170c786f6db8'
Submodule path 'vendor/github.com/jteeuwen/go-bindata': checked out 'a0ff2567cfb70903282db057e799fd826784d41d'
Submodule path 'vendor/github.com/la5nta/wl2k-go': checked out '26c05c4b44a13651c240a3946d70ef8bd9aba8b7'
Submodule 'fbb/vendor/github.com/paulrosania/go-charset' (https://github.com/paulrosania/go-charset.git) registered for path 'vendor/github.com/la5nta/wl2k-go/fbb/vendor/github.com/paulrosania/go-charset'
Submodule 'transport/ax25/vendor/github.com/tarm/goserial' (https://github.com/tarm/goserial.git) registered for path 'vendor/github.com/la5nta/wl2k-go/transport/ax25/vendor/github.com/tarm/goserial'
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/la5nta/wl2k-go/fbb/vendor/github.com/paulrosania/go-charset'...
Cloning into '/Users/uri/src/gosrc/pat/vendor/github.com/la5nta/wl2k-go/transport/ax25/vendor/github.com/tarm/goserial'...
Submodule path 'vendor/github.com/la5nta/wl2k-go/fbb/vendor/github.com/paulrosania/go-charset': checked out '621bb39fcc835dce592e682f5073025d0169587b'
Submodule path 'vendor/github.com/la5nta/wl2k-go/transport/ax25/vendor/github.com/tarm/goserial': checked out 'b3440c3c63556c3fd67e915be3b92195bc70294a'
Submodule path 'vendor/github.com/mattn/go-runewidth': checked out '97311d9f7767e3d6f422ea06661bc2c7a19e8a5d'
Submodule path 'vendor/github.com/microcosm-cc/bluemonday': checked out 'e79763773ab6222ca1d5a7cbd9d62d83c1f77081'
Submodule path 'vendor/github.com/nfnt/resize': checked out '891127d8d1b52734debe1b3c3d7e747502b6c366'
Submodule path 'vendor/github.com/peterh/liner': checked out '88609521dc4b6c858fd4c98b628147da928ce4ac'
Submodule path 'vendor/github.com/spf13/pflag': checked out 'e57e3eeb33f795204c1ca35f56c44f83227c6e66'
Submodule path 'vendor/golang.org/x/crypto': checked out '6914964337150723782436d56b3f21610a74ce7b'
Submodule path 'vendor/golang.org/x/net': checked out 'ab5485076ff3407ad2d02db054635913f017b0ed'
Submodule path 'vendor/golang.org/x/sys': checked out 'c4489faa6e5ab84c0ef40d6ee878f7a030281f0f'
Submodule path 'vendor/golang.org/x/text': checked out '836efe42bb4aa16aaa17b9c155d8813d336ed720'
Running tests...
can't load package: package _/Users/uri/src/gosrc/pat: cannot find package "_/Users/uri/src/gosrc/pat" in any of:
	/opt/local/lib/go/src/_/Users/uri/src/gosrc/pat (from $GOROOT)
	/Users/uri/src/gosrc/src/_/Users/uri/src/gosrc/pat (from $GOPATH)
can't load package: package _/Users/uri/src/gosrc/pat/cfg: cannot find package "_/Users/uri/src/gosrc/pat/cfg" in any of:
	/opt/local/lib/go/src/_/Users/uri/src/gosrc/pat/cfg (from $GOROOT)
	/Users/uri/src/gosrc/src/_/Users/uri/src/gosrc/pat/cfg (from $GOPATH)
can't load package: package _/Users/uri/src/gosrc/pat/internal/cmsapi: cannot find package "_/Users/uri/src/gosrc/pat/internal/cmsapi" in any of:
	/opt/local/lib/go/src/_/Users/uri/src/gosrc/pat/internal/cmsapi (from $GOROOT)
	/Users/uri/src/gosrc/src/_/Users/uri/src/gosrc/pat/internal/cmsapi (from $GOPATH)
can't load package: package _/Users/uri/src/gosrc/pat/internal/gpsd: cannot find package "_/Users/uri/src/gosrc/pat/internal/gpsd" in any of:
	/opt/local/lib/go/src/_/Users/uri/src/gosrc/pat/internal/gpsd (from $GOROOT)
	/Users/uri/src/gosrc/src/_/Users/uri/src/gosrc/pat/internal/gpsd (from $GOPATH)
$
```

I'm not a GO expert, and frankly am not looking forward to becoming one. Would appreciate help getting this app installed and running.

Also, there doesn't seem to be a command `packagesbuild`:
```
$ command -v packagesbuild
$ type packagesbuild
-bash: type: packagesbuild: not found
$ 
```

**Comment by martinhpedersen** [comment]
> Then I created directory $HOME/src/gosrc and did export GOPATH=$HOME/src/gosrc. Then I did cd $GOPATH; go get -u github.com/la5nta/pat.

Could you try this again, but without the -u flag? (https://github.com/davecheney/httpstat/issues/120)

Out of curiosity, what version of Go and git are you running?

> Attempt to manually clone and build also failed.

If you want to manually clone the repo, then make sure to clone it into `$GOPATH/src/github.com/la5nta/pat`, `not $GOPATH/src/pat`.

> I'm not a GO expert, and frankly am not looking forward to becoming one. Would appreciate help getting this app installed and running.

Hope this helps, and please keep an open mind :-)

> Also, there doesn't seem to be a command packagesbuild:

Yes, that is correct. If you want to build the pkg installer package, then you'll need to install [Packages](http://s.sudre.free.fr/Software/Packages/about.html). I guess it is not mentioned on the wiki page, but the make.bash script is mainly there for the developers convenience when building the official binary releases.

PS: You could also grab the latest pre-built binary for macOS. It's available from the [releases page](https://github.com/la5nta/pat/releases).

**Comment by martinhpedersen** [comment]
https://github.com/golang/go/issues/17522

**Comment by martinhpedersen** [comment]
12. jan. 2018 18:44 skrev "Mouse" <notifications@github.com>:

I installed go via Macports (it installs stuff to /opt/local) and set export
GOROOT=/opt/local/lib/go.

Then I created directory $HOME/src/gosrc and did export
GOPATH=$HOME/src/gosrc. Then I did cd $GOPATH; go get -u
github.com/la5nta/pat. Result:

$ go get -u github.com/la5nta/pat
package github.com/la5nta/pat/vendor/github.com/bndr/gotabulate:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/bndr/gotabulate/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/elazarl/go-bindata-assetfs:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/elazarl/go-bindata-assetfs/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/fsnotify/fsnotify:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/fsnotify/fsnotify/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/gorhill/cronexpr:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/gorhill/cronexpr/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/gorilla/mux:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/gorilla/mux/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/gorilla/websocket:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/gorilla/websocket/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/howeyc/gopass:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/howeyc/gopass/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/transport/ardop:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/fbb/vendor/github.com/paulrosania/go-charset/charset:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/fbb/vendor/github.com/paulrosania/go-charset/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/transport/ax25/vendor/github.com/tarm/goserial:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/la5nta/wl2k-go/transport/ax25/vendor/github.com/tarm/goserial/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/microcosm-cc/bluemonday:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/microcosm-cc/bluemonday/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/nfnt/resize:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/nfnt/resize/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/peterh/liner:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/peterh/liner/.git
exists but is not a directory
package github.com/la5nta/pat/vendor/github.com/spf13/pflag:
/Users/uri/src/gosrc/src/github.com/la5nta/pat/vendor/github.com/spf13/pflag/.git
exists but is not a directory
$


Could you try this again, but without the -u flag?

https://github.com/davecheney/httpstat/issues/120

Out of curiosity, what version of Go and git are you running?

 Attempt to manually clone and build also failed:

$ go env gopath
$ echo $GOROOT
/opt/local/lib/go
$ echo $GOPATH
/Users/uri/src/gosrc
$ pwd
/Users/uri/src/gosrc
$ ls
src/
$ git clone https://github.com/la5nta/pat


If you want to manually clone the repo, then make sure to clone it into
$GOPATH/src/github.com/la5nta/pat, not $GOPATH/src/pat.

 I'm not a GO expert, and frankly am not looking forward to becoming one.
Would appreciate help getting this app installed and running.


Hope this helps, and please keep an open mind :-)

PS: You could also grab the latest pre-built binary for macOS. It's
available from the releases page at github.

-- 
Martin

**Comment by mouse07410** [comment]
> Out of curiosity, what version of Go and git are you running?

I think it's 1.9.2.

```
$ echo $GOPATH
/Users/uri/src/gosrc
$ echo $GOROOT
/opt/local/lib/go
$ go version
go version go1.9.2 darwin/amd64
$ pwd
/Users/uri/src/gosrc
$ uname -a
Darwin Uris-MacBook-Air.local 17.3.0 Darwin Kernel Version 17.3.0: Thu Nov  9 18:09:22 PST 2017; root:xnu-4570.31.3~1/RELEASE_X86_64 x86_64
$ ls -l
$ ls -al
total 0
drwxr-xr-x    2 uri  staff     64 Jan 12 17:04 .
drwxr-xr-x  798 uri  staff  25536 Jan 11 20:59 ..
$ 
```

> Could you try this again, but without the -u flag? (davecheney/httpstat#120)

Funny - I tried, and it seemed to work:

```
$ go get github.com/la5nta/pat
$ ll
total 0
drwxr-xr-x    5 uri  staff    160 Jan 12 17:08 ./
drwxr-xr-x  798 uri  staff  25536 Jan 11 20:59 ../
drwxr-xr-x    3 uri  staff     96 Jan 12 17:08 bin/
drwxr-xr-x    3 uri  staff     96 Jan 12 17:07 pkg/
drwxr-xr-x    3 uri  staff     96 Jan 12 17:07 src/
$ ll bin
total 29056
drwxr-xr-x  3 uri  staff        96 Jan 12 17:08 ./
drwxr-xr-x  5 uri  staff       160 Jan 12 17:08 ../
-rwxr-xr-x  1 uri  staff  14874412 Jan 12 17:08 pat*
$ bin/pat version
Pat v0.5.1 (unknown origin) darwin/amd64 - go1.9.2
$
```

> Hope this helps, and please keep an open mind :-)

Yes indeed, it appeared to resolve the problem. Thank you!

And yes, I do keep an open mind - but there are more languages than I can hope to learn sufficiently well, and more build environments than fleas on a mangy dog - so again, I don't expect to become proficient in those that I'm not likely to use on a semi-daily/weekly/monthly basis :-)

Again, thank you! Losing `-u` turned failure to success. ;-)

**Comment by martinhpedersen** [comment]
Glad it worked out! Let's hope the issue is fixed in Go 1.10, which is currently in beta.

I will review the build instructions on the wiki to make sure they are up to date and correct.

73
